commodore/catalog.py

```diff
@@ -65,15 +66,15 @@ def clean_catalog(repo):
         rm_tree_contents(repo.working_tree_dir)
 
 
-def update_catalog(cfg, target_name, repo):
+def update_catalog(cfg, targets: Iterable[str], repo):
     click.secho("Updating catalog repository...", bold=True)
     # pylint: disable=import-outside-toplevel
     from distutils import dir_util
     import textwrap
 
     catalogdir = P(repo.working_tree_dir, "manifests")
-    # copy compiled catalog into catalog directory
-    dir_util.copy_tree(P("compiled") / target_name, str(catalogdir))
+    for target_name in targets:
+        dir_util.copy_tree(str(P("compiled") / target_name), str(catalogdir))
```

Comment:
```suggestion
        dir_util.copy_tree(str(P("compiled") / target_name), str(catalogdir / target_name))
```
If we change this to copy stuff to a subdir named by the component, we don't have to specify the component name in the various `compile` items anymore:

```
      - output_path: argocd/00_crds/
        input_type: copy
        input_paths:
          - argocd/manifests/${argocd:git_tag}/crds/
```
becomes
```
      - output_path: 00_crds/
        input_type: copy
        input_paths:
          - manifests/${argocd:git_tag}/crds/
```
If we also change the search path to `dependencies/$component_name` the `input_paths` also become nicer.

Reply:
It would need some special handling for the `apps` part since all Argo CD apps need to go into the same directory (not by component).

Reply:
IMHO it would make components much simpler (and more similar to all the Kapitan examples) but would require some migration path, I guess.
Other opinions?

Reply:
> If we also change the search path to `dependencies/$component_name` the `input_paths` also become nicer.

This will continue to explode if there's multiple components with identically named files, as we still only invoke `kapitan.compile_targets()` a single time, but now with a list of targets.

Reply:
Hmm true. Well in that case we might as well leave it as is. At least it is consistent between `output_path` and `input_paths`.